splunk-otel-android/src/main/java/com/splunk/rum/InitializationEvents.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright Splunk Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.splunk.rum;
+
+import static com.splunk.rum.SplunkRum.COMPONENT_APPSTART;
+import static com.splunk.rum.SplunkRum.COMPONENT_KEY;
+
+import androidx.annotation.NonNull;
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.api.trace.Tracer;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.rum.internal.instrumentation.startup.AppStartupTimer;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+class InitializationEvents {
+    private final AppStartupTimer startupTimer;
+    private final List<Event> events = new ArrayList<>();
+
+    InitializationEvents(AppStartupTimer startupTimer) {
+        this.startupTimer = startupTimer;
+    }
+
+    void emit(String eventName) {
+        events.add(new Event(eventName, startupTimer.clockNow()));
+    }
+
+    void recordInitializationSpans(long startTimeNanos, ConfigFlags flags, Tracer delegateTracer) {
+        Tracer tracer =
+                spanName ->
+                        delegateTracer
+                                .spanBuilder(spanName)
+                                .setAttribute(COMPONENT_KEY, COMPONENT_APPSTART);
+
+        Span overallAppStart = startupTimer.start(tracer);
+        Span span =
+                tracer.spanBuilder("SplunkRum.initialize")
+                        .setParent(Context.current().with(overallAppStart))
+                        .setStartTimestamp(startTimeNanos, TimeUnit.NANOSECONDS)
+                        .setAttribute(COMPONENT_KEY, COMPONENT_APPSTART)
+                        .startSpan();
+
+        String configSettings = buildConfigSettings(flags);
+        span.setAttribute("config_settings", configSettings);
+
+        for (Event initializationEvent : events) {
+            span.addEvent(initializationEvent.name, initializationEvent.time, TimeUnit.NANOSECONDS);
+        }
+        long spanEndTime = startupTimer.clockNow();
+        // we only want to create SplunkRum.initialize span when there is a AppStart span so we
+        // register a callback that is called right before AppStart span is ended
+        startupTimer.setCompletionCallback(() -> span.end(spanEndTime, TimeUnit.NANOSECONDS));
+    }
+
+    @NonNull
+    private String buildConfigSettings(ConfigFlags flags) {
```

Comment:
Nit: WDYT about putting all this into `ConfigFlags#toString()`?

Reply:
I like it!